src/twisted/internet/testing.py

```diff
@@ -967,3 +967,54 @@ def createWithCleanup(cls, testInstance: TestCase, publisher: LogPublisher) -> S
         publisher.addObserver(obs)
         testInstance.addCleanup(lambda: publisher.removeObserver(obs))
         return obs
+
+
+def benchmarkWithReactor(test_target):
```

Comment:
So I just remembered that there's actually a library for this: https://pypi.org/project/pytest-twisted/

So using that would probably make everything simpler.

Reply:
Uses `greenlet` :sob:

Reply:
Can maybe stick to this, that's a little terrifying.